### PR TITLE
Dtj chapter14

### PR DESCRIPTION
**Pull request by thatguy59**

### CODE REVIEW

File: modules/dtj/My-List-Util/t/MyListShuffle.t [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    is_deeply ( My::List::Util::shuffle( ( 1 ) ), ( 1 ), "Incorrect shuffle of single element in array"  );
+    is_deeply ( My::List::Util::shuffle( 1 ), ( 1 ), "Incorrect shuffle of single element"  )
+};
+
+subtest 'Bad input' => sub {
+    plan tests => 1;
+    test_error( sub {  My::List::Util::shuffle() }, "Input was empty", "no input" );
+};
+
+sub test_error {
```
Comment (a-panayotopoulos) [inline modules/dtj/My-List-Util/t/MyListShuffle.t]: Good stuff. There is however already a module on CPAN for this.

File: modules/dtj/My-List-Util/t/MyListShuffle.t [hunk 2]
```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/perl
+
+use strict;
+use warnings;
+
+use Test::More tests => 5;
+
+# It would be nice to give up if either of these two fail
+use_ok( 'My::List::Util' );
+can_ok( 'My::List::Util', "shuffle" );
```
Comment (a-panayotopoulos) [inline modules/dtj/My-List-Util/t/MyListShuffle.t]: This is a lot nicer than the book's suggestion! Good stuff.

File: modules/dtj/My-List-Util/t/MyListShuffle.t (hunk 1 above)
Comment (thatguy59) [inline modules/dtj/My-List-Util/t/MyListShuffle.t]: Yeah I remember seeing/using it - I thought I'd go for the books way of doing it for no good reason.

File: modules/dtj/My-List-Util/.gitignore [hunk 3]
```diff
@@ -0,0 +1,17 @@
+Makefile
```
Comment (thatguy59) [inline modules/dtj/My-List-Util/.gitignore]: Possibly interesting point here: This `.gitignore` was created by `module_starter` by providing the arg `--ignores=git`

File: modules/dtj/My-List-Util/lib/My/List/Util.pm [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+  return @output;
+}
+
+=head1 AUTHOR
+
+Daniel jones, C<< <dtj at someplace.com> >>
+
+=head1 BUGS
+
+Included free of charge
```
Comment (a-panayotopoulos) [inline modules/dtj/My-List-Util/lib/My/List/Util.pm]: :laughing:

**Pull request merged**